src/builder/QuarkBuilder.sol

```diff
@@ -1308,6 +1308,262 @@ contract QuarkBuilder {
+                chainId: supplyIntent.chainId,
+                sender: supplyIntent.sender
+            }),
+            payment,
+            useQuotecall
+        );
+
+        List.addQuarkOperation(quarkOperations, supplyQuarkOperation);
+        List.addAction(actions, supplyAction);
+
+        // Convert actions and quark operations to array
+        Actions.Action[] memory actionsArray = List.toActionArray(actions);
+        IQuarkWallet.QuarkOperation[] memory quarkOperationsArray = List.toQuarkOperationArray(quarkOperations);
+
+        // Validate generated actions for affordability
+        if (payment.isToken) {
+            assertSufficientPaymentTokenBalances(
+                actionsArray, chainAccountsList, supplyIntent.chainId, supplyIntent.sender
+            );
+        }
+
+        // Merge operations that are from the same chain into one Multicall operation
+        (quarkOperationsArray, actionsArray) =
+            QuarkOperationHelper.mergeSameChainOperations(quarkOperationsArray, actionsArray);
+
+        // Wrap operations around Paycall/Quotecall if payment is with token
+        if (payment.isToken) {
+            quarkOperationsArray = QuarkOperationHelper.wrapOperationsWithTokenPayment(
+                quarkOperationsArray, actionsArray, payment, useQuotecall
+            );
+        }
+
+        return BuilderResult({
+            version: VERSION,
+            actions: actionsArray,
+            quarkOperations: quarkOperationsArray,
+            paymentCurrency: payment.currency,
+            eip712Data: EIP712Helper.eip712DataForQuarkOperations(quarkOperationsArray, actionsArray)
+        });
+    }
+
+    struct MorphoVaultWithdrawIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        uint256 chainId;
+        address withdrawer;
+    }
+
+    function morphoVaultWithdraw(
+        MorphoVaultWithdrawIntent memory withdrawIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory) {
+        // XXX confirm that you actually have the amount to withdraw
+
+        bool isMaxWithdraw = withdrawIntent.amount == type(uint256).max;
+        bool useQuotecall = false; // never use Quotecall
+        List.DynamicArray memory actions = List.newList();
+        List.DynamicArray memory quarkOperations = List.newList();
+
+        // when paying with tokens, you may need to bridge the payment token to cover the cost
+        if (payment.isToken) {
+            uint256 maxCostOnDstChain = PaymentInfo.findMaxCost(payment, withdrawIntent.chainId);
+            // if you're withdrawing the payment token, you can use the withdrawn amount to cover the cost
+            if (Strings.stringEqIgnoreCase(payment.currency, withdrawIntent.assetSymbol)) {
+                // XXX in the withdrawMax case, use the Comet balance
```

Comment:
Morpho, not Comet, I think.

also, do we need to implement this or is it a note for something to do later in a follow-up...?

Reply:
iirc hmm  I think we should remove this as Morpho script handles withdrawMax already. 
Hmm I think we should remove this from Comet's builder function too.... cuz iirc comet supports withdrawMax 🤔 already.